salt/modules/zabbix.py

```diff
@@ -2503,3 +2503,187 @@ def configuration_import(config_file, rules=None, file_format='xml', **connectio
+
+    :param hostid: ID of the host which's trigger we want to find
+    :param trigger_desc: Description of trigger (trigger name) which's we want to find
+    :param priority: Priority of trigger (useful if we have same name for more triggers with different priorities)
+
+    :param _connection_user: Optional - zabbix user (can also be set in opts or pillar, see module's docstring)
+    :param _connection_password: Optional - zabbix password (can also be set in opts or pillar, see module's docstring)
+    :param _connection_url: Optional - url of zabbix frontend (can also be set in opts, pillar, see module's docstring)
+
+    :return: Trigger ID and description. False if no trigger found or on failure.
+
+    CLI Example:
+    .. code-block:: bash
+
+        salt '*' zabbix.triggerid_get 1111 'trigger name to find' 5
+    '''
+    conn_args = _login(**connection_args)
+    ret = False
+    try:
+        if conn_args:
+            method = 'trigger.get'
+            if not hostid or not trigger_desc:
+                return {'result': False, 'comment': 'hostid and trigger_desc params are required'}
+            params = {'output': ["triggerid", "description"],
+                      'filter': {"priority": priority}, 'hostids': hostid}
+            params = _params_extend(params, _ignore_name=True, **connection_args)
+            ret = _query(method, params, conn_args['url'], conn_args['auth'])
+            if len(ret['result']) > 0:
+                for r in ret['result']:
+                    if trigger_desc in r['description']:
+                        ret['result'] = r
+                        return ret
+                return False
+            else:
+                return False
+        else:
+            raise KeyError
+    except KeyError:
+        return ret
+
+
+def service_add(service_rootid=None, service_name=None, triggerid=None, **connection_args):
+    '''
+    .. versionadded:: Fluorine
+
+    Create service under service with id specified as parameter.
+
+    .. note::
+        https://www.zabbix.com/documentation/3.4/manual/api/reference/service/create
+
+    :param service_rootid: Service id under which service should be added
+    :param service_name: Name of new service
+    :param triggerid: Optional - ID of trigger which should be watched in service
+
+    :param _connection_user: Optional - zabbix user (can also be set in opts or pillar, see module's docstring)
+    :param _connection_password: Optional - zabbix password (can also be set in opts or pillar, see module's docstring)
+    :param _connection_url: Optional - url of zabbix frontend (can also be set in opts, pillar, see module's docstring)
+
+    :return: Service details, False if service could not be added or on failure.
+
+    CLI Example:
+    .. code-block:: bash
+
+        salt '*' zabbix.service_add 11 'My service' 11111
+    '''
+    conn_args = _login(**connection_args)
+    ret = False
```

Comment:
It looks like here and in other places we're returning either a boolean or a dictionary. Should it instead be an empty dict or a popualated dict? That might avoid type errors for consumers of this module.

Reply:
good idea. I changed it in whole module.